### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by devos50**

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

File: Tribler/Core/APIImplementation/LaunchManyCore.py [hunk 1]
```diff
@@ -500,10 +500,9 @@ def _invoke_states_cb(self, callback):
             ds = d.network_get_state(None)
             dslist.append(ds)
 
-        def on_cb_done(new_get_peer_list):
-            self.get_peer_list = new_get_peer_list
-
-        return deferToThread(callback, dslist).addCallback(on_cb_done)
+        new_get_peer_list = callback(dslist)
```
Comment (egbertbouman) [inline Tribler/Core/APIImplementation/LaunchManyCore.py]: Putting the session callback on the reactor thread makes sense, but are we sure we also want to execute the 38000 SQL statements from the bootstrap file on the reactor thread? It takes 3-5 seconds on my workstation.

File: Tribler/Test/Core/Modules/test_versioncheck.py [hunk 2]
```diff
@@ -56,6 +57,20 @@ def assert_new_version_called(self, _):
...
+    def test_start(self):
+        """
+        Test whether the periodic version lookup works as expected
+        """
+        self.setup_version_server(json.dumps({'name': 'v1.0'}))
+        self.session.lm.version_check_manager.start()
+        self.assertFalse(self.session.lm.version_check_manager.is_pending_task_active("tribler version check"))
+
+        import Tribler.Core.Modules.versioncheck_manager as vcm
+        vcm.version_id = "7.0.0"
```
Comment (xoriole) [inline Tribler/Test/Core/Modules/test_versioncheck.py]: Would be nice to have a test for a version with GIT in the version_id here.

Comment (devos50) [reply Tribler/Test/Core/Modules/test_versioncheck.py]: Actually, line 66 checks for this. Note that the `version_id` attribute is set to `7.0.0-GIT` in the `version.py` file.

**Pull request merged**